apis/rds/v1alpha1/zz_db_cluster.go

```diff
@@ -0,0 +1,508 @@
+	// The number of days for which automated backups are retained.
+	//
+	// Default: 1
+	//
+	// Constraints:
+	//
+	//    * Must be a value from 1 to 35
+	BackupRetentionPeriod *int64 `json:"backupRetentionPeriod,omitempty"`
+	// A value that indicates that the DB cluster should be associated with the
+	// specified CharacterSet.
+	CharacterSetName *string `json:"characterSetName,omitempty"`
+	// A value that indicates whether to copy all tags from the DB cluster to snapshots
+	// of the DB cluster. The default is not to copy them.
+	CopyTagsToSnapshot *bool `json:"copyTagsToSnapshot,omitempty"`
+	// The DB cluster identifier. This parameter is stored as a lowercase string.
+	//
+	// Constraints:
+	//
+	//    * Must contain from 1 to 63 letters, numbers, or hyphens.
+	//
+	//    * First character must be a letter.
+	//
+	//    * Can't end with a hyphen or contain two consecutive hyphens.
+	//
+	// Example: my-cluster1
+	// +kubebuilder:validation:Required
+	DBClusterIdentifier *string `json:"dbClusterIdentifier"`
+	// The name of the DB cluster parameter group to associate with this DB cluster.
+	// If you do not specify a value, then the default DB cluster parameter group
+	// for the specified DB engine and version is used.
+	//
+	// Constraints:
+	//
+	//    * If supplied, must match the name of an existing DB cluster parameter
+	//    group.
+	DBClusterParameterGroupName *string `json:"dbClusterParameterGroupName,omitempty"`
+	// A DB subnet group to associate with this DB cluster.
+	//
+	// Constraints: Must match the name of an existing DBSubnetGroup. Must not be
+	// default.
+	//
+	// Example: mySubnetgroup
+	DBSubnetGroupName *string `json:"dbSubnetGroupName,omitempty"`
+	// The name for your database of up to 64 alphanumeric characters. If you do
+	// not provide a name, Amazon RDS doesn't create a database in the DB cluster
+	// you are creating.
+	DatabaseName *string `json:"databaseName,omitempty"`
+	// A value that indicates whether the DB cluster has deletion protection enabled.
+	// The database can't be deleted when deletion protection is enabled. By default,
+	// deletion protection is disabled.
+	DeletionProtection *bool `json:"deletionProtection,omitempty"`
+	// DestinationRegion is used for presigning the request to a given region.
+	DestinationRegion *string `json:"destinationRegion,omitempty"`
+	// The Active Directory directory ID to create the DB cluster in.
+	//
+	// For Amazon Aurora DB clusters, Amazon RDS can use Kerberos Authentication
+	// to authenticate users that connect to the DB cluster. For more information,
+	// see Kerberos Authentication (https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/kerberos-authentication.html)
+	// in the Amazon Aurora User Guide.
+	Domain *string `json:"domain,omitempty"`
+	// Specify the name of the IAM role to be used when making API calls to the
+	// Directory Service.
+	DomainIAMRoleName *string `json:"domainIAMRoleName,omitempty"`
```

Comment:
Could you implement the referencerse wherever possible? It makes a huge difference in UX for the users.